src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -184,6 +183,71 @@ TxQ::FeeMetrics::scaleFeeLevel(OpenView const& view,
     return baseLevel;
 }
 
+namespace detail {
+
+static
+std::pair<bool, std::uint64_t>
+sumOfFirstSquares(std::size_t x)
+{
+    // sum(n = 1->x) : n * n = x(x + 1)(2x + 1) / 6
+
+    // If x is anywhere on the order of 2^^21, it's going
+    // to completely dominate the computation and is likely
+    // enough to overflow that we're just going to assume
+    // it does. If we have anywhere near 2^^21 transactions
+    // in a ledger, this is the least of our problems.
```

Comment:
More specifically, in practice, `x` represents the number of transactions in a potential open ledger. While I don't think we'll ever possibly get to `pow(2,21)`, there is still a very large potential size for a lookup table.